### PR TITLE
refactor: repo maintenance

### PR DESCRIPTION
**Pull request by Agrendalath**
This:
1. Adds `catalog-info.yaml`.
2. Updates `.gitignore`.
3. Removes some leftover code from Python 2.
4. Modernizes building the package.
5. Modernizes running test.
6. Removes unused integration tests.
7. Updates requirements.
8. Adds automatic Python requirement upgrades.
9. Adds the CI workflow to upload the package to PyPI.
10. Improves CI scripts.

### CODE REVIEW

**Comment by openedx-webhooks** [comment]
Thanks for the pull request, @Agrendalath! Please note that it may take us up to several weeks or months to complete a review and merge your PR.

Feel free to add as much of the following information to the ticket as you can:
- supporting documentation
- [Open edX discussion forum threads](https://discuss.openedx.org/)
- timeline information ("this must be merged by XX date", and why that is)
- partner information ("this is a course on edx.org")
- any other information that can help Product understand the context for the PR

All technical communication about the code itself will be done via the GitHub pull request interface. As a reminder, [our process documentation is here](http://edx-developer-guide.readthedocs.org/en/latest/process/overview.html).


This is currently a draft pull request. When it is ready for our review and all tests are green, click "Ready for Review", or remove "WIP" from the title, as appropriate.

File: .github/workflows/pypi-publish.yml [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+        run: pip install -r requirements/pip.txt
+
+      - name: Build package
+        run: python setup.py sdist bdist_wheel
+
+      - name: Publish to PyPi
+        uses: pypa/gh-action-pypi-publish@release/v1
+        with:
+          user: __token__
+          password: ${{ secrets.PYPI_UPLOAD_TOKEN }}
```
Comment (Agrendalath) [inline .github/workflows/pypi-publish.yml:30]: @feanil, do we need to add this secret to this repo explicitly, or is it shared?

**Comment by Agrendalath** [comment]
@feanil, would you mind reviewing this? I tried to make the git history easy to follow.

**Comment by feanil** [comment]
Yes, I'll take a look tomorrow.

**Review by feanil — APPROVED** [review]
Amazing work @Agrendalath , the changes all look good to me!

**Comment by openedx-webhooks** [comment]
@Agrendalath 🎉 Your pull request was merged!  Please take a moment  to [answer a two question survey](https://docs.google.com/forms/d/e/1FAIpQLSceJOyGJ6JOzfy6lyR3T7EW_71OWUnNQXp68Fymsk3MkNoSDg/viewform?usp=pp_url&entry.1671973413=openedx/xblock-image-explorer&entry.867055334=https://github.com/openedx/xblock-image-explorer/pull/109&entry.1484655318=https://github.com/Agrendalath&entry.752974735=2024-01-17+02:31&entry.1917517419=2024-01-18+14:39&entry.2133058324=Yes) so we can improve your experience in the future.

File: .github/workflows/pypi-publish.yml (hunk 1 above)
Comment (Agrendalath) [reply .github/workflows/pypi-publish.yml:30]: Right, it worked correctly: https://pypi.org/project/xblock-image-explorer/.

**Comment by Agrendalath** [comment]
@feanil, [the CI fails](https://github.com/openedx/xblock-image-explorer/actions/runs/7604793814/job/20708155372) to [create a git tree](https://docs.github.com/en/rest/git/trees?apiVersion=2022-11-28#create-a-tree) in this repo (and in the `openedx/xblock-google-drive`). Are some additional permissions required for the requirements bot?

**Comment by feanil** [comment]
Yes, there was a team that needed to be added, I've done so and verified that it is fixed.

File: setup.py [hunk 2]
```diff
@@ -19,19 +96,35 @@ def package_data(pkg, root_list):
...
 
 setup(
     name='xblock-image-explorer',
-    version='2.1.0',
+    version=VERSION,
     description='XBlock - Image Explorer',
-    packages=['image_explorer'],
-    install_requires=[
-        'XBlock>=1.2',
-        'parsel>=1.6.0,<=1.6.0',
```
Comment (0x29a) [inline setup.py:31]: @Agrendalath, FYI: it looks like removing of this constraint breaks this XBlock for Quince. Consider this traceback that I'm getting when trying to add it to a unit:
```python
2024-01-29 07:31:31,744 WARNING 13 [xblock.plugin] [user 112] [ip 91.196.101.94] plugin.py:144 - Unable to load XBlock 'image-explorer'                                                                                                                                                     
Traceback (most recent call last):                                                                                                                                                                                                                                                          
  File "/openedx/venv/lib/python3.8/site-packages/xblock/plugin.py", line 141, in load_classes                                                                                                                                                                                              
    yield (class_.name, cls._load_class_entry_point(class_))                                                                                                                                                                                                                                
  File "/openedx/venv/lib/python3.8/site-packages/xblock/plugin.py", line 70, in _load_class_entry_point                                                                                                                                                                                    
    class_ = entry_point.load()                                                                                                                                                                                                                                                             
  File "/openedx/venv/lib/python3.8/site-packages/pkg_resources/__init__.py", line 2518, in load                                                                                                                                                                                            
    return self.resolve()                                                                                                                                                                                                                                                                   
  File "/openedx/venv/lib/python3.8/site-packages/pkg_resources/__init__.py", line 2524, in resolve                                                                                                                                                                                         
    module = __import__(self.module_name, fromlist=['__name__'], level=0)                                                                                                                                                                                                                   
  File "/openedx/venv/lib/python3.8/site-packages/image_explorer/__init__.py", line 23, in <module>                                                                                                                                                                                         
    from .image_explorer import ImageExplorerBlock                                                                                                                                                                                                                                          
  File "/openedx/venv/lib/python3.8/site-packages/image_explorer/image_explorer.py", line 37, in <module>                                                                                                                                                                                   
    from parsel import Selector                                                                                                                                                                                                                                                             
  File "/openedx/venv/lib/python3.8/site-packages/parsel/__init__.py", line 16, in <module>                                                                                                                                                                                                 
    from parsel.selector import Selector, SelectorList  # NOQA                                                                                                                                                                                                                              
  File "/openedx/venv/lib/python3.8/site-packages/parsel/selector.py", line 48, in <module>                                                                                                                                                                                                 
    lxml_version = Version(etree.__version__)                                                                                                                                                                                                                                               
AttributeError: module 'openedx.core.lib.safe_lxml.etree' has no attribute '__version__'                                                                                                                                                                                                    
```

`etree.__version__` has been added in [parsel==1.7.0](https://github.com/scrapy/parsel/commit/f9db6d9aea0a99c95338333fffd497c6794562d0).

Comment (Agrendalath) [reply setup.py:31]: Thanks for catching this, @0x29a! This is now fixed in [v2.2.1](https://github.com/openedx/xblock-image-explorer/releases/tag/v2.2.1).

Comment (0x29a) [reply setup.py:31]: Thanks for fixing this so quickly, @Agrendalath!

**Pull request merged**